### PR TITLE
Collision refactor

### PR DESCRIPTION
**Pull request by Sindex42**
Moved collision to its own module. Updated hero and enemy classes accordingly

### CODE REVIEW

**Review by matharotheelf — APPROVED** [review]
Great refactor!

File: src/collision.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+''' Collision module '''
+
+import os
+import pygame as pg
+
+def collide(subject, sprites, d_x=0, d_y=0, callback=None):
```
Comment (CKKH) [inline src/collision.py:6]: Awesome to see callback in action with Python

**Review by CKKH — APPROVED** [review]
Really clever refactoring, good job.

File: src/collision.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    ''' Check for collision '''
+
+    for sprite in sprites:
+        if sprite.x_pos == subject.x_pos + d_x and sprite.y_pos == subject.y_pos + d_y:
+            if callback is not None:
```
Comment (michaelnguyen974) [inline src/collision.py:11]: Really well done on this

File: src/collision.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+''' Collision module '''
```
Comment (michaelnguyen974) [inline src/collision.py:1]: Very good abstraction

**Review by michaelnguyen974 — APPROVED** [review]

**Pull request merged**